packages/core/src/external/sftp/id-generator.ts

```diff
@@ -0,0 +1,57 @@
+export type IdGenerator = (time?: number, entropy?: 0 | 1) => Buffer;
+const LEXICON = "-0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz";
+const LEXICON_START = LEXICON.charCodeAt(0);
+const LEXICON_END = LEXICON.charCodeAt(LEXICON.length - 1);
+
+export function createIdGenerator(totalLength: number): IdGenerator {
```

Comment:
The rationale for this is that we get 10 bytes to provide a unique transaction ID, and this method (which I've tested with the SS sandbox) does work for producing something that might be described as a "10 byte UUID v7". That is, the generated ID is guaranteed to be

- unique, even if generated in the same millisecond (there are 2 bytes of randomness after the 8 bytes of timestamp)
- allowed by Surescripts' ASCII range for IDs, which is 32 - 126 (see `LEXICON`)
- lexicographically sortable (i.e. an ID which was issued later will always be alphabetically after an ID which was issued sooner, and vice versa)
- synchronous (vs having to add an asynchronous DB request)